src/bindings/python/flux/job/Jobspec.py

```diff
@@ -741,6 +741,61 @@ def _v1_validate(resources, tasks, kwargs):
             if task["count"].keys().isdisjoint({"per_slot", "total"}):
                 raise ValueError("count per_slot or total must be set")
 
+    def _set_extra_options(
+        self,
+        duration=None,
+        environment=None,
+        env_expand=None,
+        cwd=None,
+        rlimits=None,
+        name=None,
+        stdin=None,
+        stdout=None,
+        stderr=None,
+        label_io=False,
+        unbuffered=False,
+        queue=None,
+        bank=None,
+    ):
+        """Process common jobspec factory args and return amended jobspec"""
+        if duration is not None:
+            self.duration = duration
+        if environment is not None:
+            self.environment = environment
+        if env_expand is not None:
+            if not isinstance(env_expand, abc.Mapping):
+                raise ValueError("env_expand must be a mapping")
+            self.setattr_shell_option("env-expand", env_expand)
+        if cwd is not None:
+            self.cwd = cwd
+        if rlimits is not None:
+            if not isinstance(rlimits, abc.Mapping):
+                raise ValueError("rlimits must be a mapping")
+            self.setattr_shell_option("rlimit", rlimits)
+        if name is not None:
+            self.setattr("job.name", name)
+        if stdin is not None:
+            self.stdin = stdin
+        if stdout is not None and stdout not in ("none", "kvs"):
+            self.stdout = stdout
+            if label_io:
+                self.setattr_shell_option("output.stdout.label", True)
+        if stderr is not None and stderr not in ("none", "kvs"):
+            self.stderr = stderr
+            if label_io:
+                self.setattr_shell_option("output.stderr.label", True)
+        if unbuffered:
+            #  For output, set the buffer.type to none and reduce the configured
+            #  event batch-timeout to something very small.
+            self.setattr_shell_option("output.stdout.buffer.type", "none")
+            self.setattr_shell_option("output.stderr.buffer.type", "none")
+            self.setattr_shell_option("output.batch-timeout", 0.05)
+        if queue is not None:
+            self.setattr("queue", queue)
+        if bank is not None:
+            self.setattr("bank", bank)
```

Comment:
Since you're adding some flux-accounting attributes, you could consider adding `project` (the flux-accounting implementation of "WCKeys")  as another option here too, but not a huge deal either way; I'm not sure how much use projects get these days. I can't remember if it has to be an official jobspec v1 property to be listed here, though, so if so, feel free to ignore me.

Reply:
I think we were just trying to add options that are supported in the submission CLI commands :shrug:. Since the submission commands support `--bank` and `--queue`, I've added those here. However, I did leave other stuff out somewhat arbitrarily so I'm open to other ideas.